docker/rockylinux9/Dockerfile

```diff
@@ -1,7 +1,10 @@
-FROM rockylinux:9.3 AS build
+FROM docker.io/rockylinux/rockylinux:9.6 AS build
```

Comment:
Not sure if we want to add `docker.io/` prefix, since it will probably disable the docker cache and cause build failures if we pull the image more than a few times a day. Probably we should update our internal docker-cache specs to include 9.6 first.

Another question is: will 9.6 compiled binary work for our RHEL 9 machines?